src/pybind/mgr/balancer/module.py

```diff
@@ -1220,8 +1220,13 @@ def do_crush_compat(self, plan: MsPlan) -> Tuple[int, str]:
                                 'change balancer mode and retry might help'
 
     def get_compat_weight_set_weights(self, ms: MappingState):
-        if not CRUSHMap.have_default_choose_args(ms.crush_dump):
+        have_choose_args = CRUSHMap.have_default_choose_args(ms.crush_dump)
+        if have_choose_args:
+            # get number of buckets in choose_args
+            choose_args_len = len(CRUSHMap.get_default_choose_args(ms.crush_dump))
```

Comment:
Why not do this here?

choose_args_len = len(have_choose_args)

Reply:
```
    def have_default_choose_args(dump: Dict[str, Any]) -> bool:
        return CRUSHMap.DEFAULT_CHOOSE_ARGS in dump.get('choose_args', {})

    @staticmethod
    def get_default_choose_args(dump: Dict[str, Any]) -> List[Dict[str, Any]]:
        choose_args = dump.get('choose_args')
        assert isinstance(choose_args, dict)
        return choose_args.get(CRUSHMap.DEFAULT_CHOOSE_ARGS, [])
```